moped-database/migrations/1698950636188_create_shared_use_trail_dual_path_component/up.sql

```diff
@@ -0,0 +1,43 @@
+INSERT INTO moped_components (
+    component_name,component_subtype,line_representation,feature_layer_id
+) VALUES
+    ('Trail', 'Shared Use Path (Paved Dual Trail)', TRUE, 4);
+
+-- insert all "protection type" subcomponents into moped_components_subcomponents
+WITH inserts_todo AS (
+    SELECT
+        subcomponent_id,
+        'Trail' AS component_name,
+        'Shared Use Path (Paved Dual Trail)' AS component_subtype
+    FROM
+        moped_subcomponents
+    WHERE
+        subcomponent_name LIKE 'Protection Type%'
+) INSERT INTO moped_components_subcomponents (component_id, subcomponent_id)
+SELECT
+    mc.component_id, inserts_todo.subcomponent_id
+FROM
+    inserts_todo
+    -- gets the component id of the new component we created
+    LEFT JOIN moped_components mc ON mc.component_name = inserts_todo.component_name
+        AND mc.component_subtype = inserts_todo.component_subtype;
+
+-- insert the same moped_component_work_types work types as the Trail - Shared Use Path (Paved) component
```

Comment:
this is a kinda fancy way of doing this that we probably want to bookmark. it copies the subcomponent and work types of another similar component.

also, it does not rely on us knowing the `component_id` of the component we just created ☝️ instead we get that component ID by query on the component name + subtype.

Reply:
this is really nice, and I agree! 🔖